File: python/character_match.py
# Don't forget to run the tests (and create some of your own)

# Part 1
def is_character_match(string1, string2):
    list1 = [*string1.replace(' ', '').lower()]
    list2 = [*string2.replace(' ', '').lower()]
    
    if len(list1) != len(list2):
        return False
    else:
        chars_found = 0
        for char in list1:
            if list2.index(char) >= 0:
                chars_found += 1
        if chars_found == len(list1):
            return True
        else:
            return False
            


# Part 2
list_of_words = ["threads", "trashed", "hardest", "hatreds", "hounds"]
def anagrams_for(word, list_of_words):
    anagrams = []
    for curr in list_of_words:
        if is_character_match(word, curr) == True:
            anagrams.append(curr)
    return anagrams

print(anagrams_for("threads", list_of_words))

File: js/characterMatch.js
exports.isCharacterMatch = function(string1, string2) {

};

exports.anagramsFor = function(word, listOfWords) {

};

function isCharacterMatch(string1, string2) {
    const arr1 = string1.replace(/\s/g,'').split("");
    const arr2 = string2.replace(/\s/g,'').split("");

    if (arr1.length !== arr2.length) {
        return false;
    } else {
        charsFound = 0;
        for (let i = 0; i < arr1.length; i++) {
            let char = arr1[i];
            if (arr2.indexOf(char.toLowerCase()) >= 0 || arr2.indexOf(char.toUpperCase() >= 0)) {
                charsFound++;
            }
        }

        if (charsFound === arr1.length) {
            return true;
        } else {
            return false;
        }
    }
}

listOfWords = ["threads", "trashed", "hardest", "hatreds", "hounds"];
function anagramsFor(word, listOfWords) {
    let anagrams = [];
    for (let i = 0; i < listOfWords.length; i++) {
        let curr = listOfWords[i]
        if (isCharacterMatch(word, curr) === true) {
            anagrams.push(curr)
        }
    }
    return anagrams
}

console.log(anagramsFor("threads", listOfWords))